distribution/launcher/src/main/java/org/eclipse/sensinact/gateway/launcher/FeatureLauncher.java

```diff
@@ -307,19 +336,29 @@ private Feature loadFeature(String feature) {
 
         LOGGER.debug("Loading feature model {}", feature);
 
-        Path featureFile;
+        Path featureFile = null;
         if (feature.indexOf(':') >= 0) {
-            featureFile = getFileFromRepository(featureService.getIDfromMavenCoordinates(feature), "json");
+            try {
+                featureFile = getFileFromRepository(featureService.getIDfromMavenCoordinates(feature), "json");
+            } catch (FileNotFoundException e) {
+                LOGGER.warn("Can't find feature file: {}", feature, e);
+            }
         } else {
             String simpleFileName = feature;
             if (!simpleFileName.endsWith(".json")) {
                 simpleFileName += ".json";
             }
 
-            featureFile = featureDir.resolve(simpleFileName);
+            for (Path featureDir : featureDirs) {
+                Path testedFile = featureDir.resolve(simpleFileName);
+                if (Files.exists(testedFile)) {
+                    featureFile = testedFile;
+                    break;
+                }
+            }
         }
 
-        if (Files.isRegularFile(featureFile)) {
+        if (featureFile != null && Files.isRegularFile(featureFile)) {
```

Comment:
If we add the isRegularFile check above then this can be
```suggestion
        if (featureFile != null) {
```

Reply:
The `getFileFromRepository` method can return an invalid path, so the isRegularFile check should stay.